saleor/webhook/transport/asynchronous/transport.py

```diff
@@ -251,13 +317,76 @@ def trigger_webhooks_async(
         )
 
 
+def _generate_deferred_payload(
+    event_type: str,
+    webhook: "Webhook",
+    payload_args: dict,
+    delivery: "EventDelivery",
+    logger_obj: "logging.Logger",
+):
+    args_obj = DeferredPayloadData(**payload_args)
+    requestor = None
+    if args_obj.requestor_model_id and args_obj.requestor_model_name in (
+        RequestorModelName.APP,
+        RequestorModelName.USER,
+    ):
+        model = apps.get_model(args_obj.requestor_model_name)
+        requestor = model.objects.filter(pk=args_obj.requestor_model_id).first()
+
+    subscribable_object = (
+        apps.get_model(args_obj.model_name).objects.filter(pk=args_obj.model_id).first()
+    )
+    if not subscribable_object:
+        logger_obj.error(
+            (
+                "[Webhook ID:%r] Failed to get the subscribable_object for deferred "
+                "payload; event: %s, model name: %s, model ID: %r, delivery ID: %r"
+            ),
+            webhook.id,
+            event_type,
+            args_obj.model_name,
+            args_obj.model_id,
+            delivery.id,
+        )
+
+    request = initialize_request(
+        requestor,
+        event_type in WebhookEventSyncType.ALL,
+        event_type=event_type,
+        allow_replica=True,
+        request_time=args_obj.request_time,
+    )
+    data_promise = generate_payload_promise_from_subscription(
+        event_type=event_type,
+        subscribable_object=subscribable_object,
+        subscription_query=webhook.subscription_query,  # type: ignore
+        request=request,
+        app=webhook.app,
+    )
+
+    if data_promise:
+        data = data_promise.get()
+        if data:
+            data_json = json.dumps({**data})
+
+            # Save the payload in the EventPayload object for later reuse in case of
+            # automatic or manual retry.
+            with allow_writer():
+                event_payload = EventPayload.objects.create_with_payload_file(data_json)
+                delivery.payload = event_payload
+                delivery.save(update_fields=["payload"])
+
+            return data_json
+    return None
```

Comment:
Shouldn't we have a test for cases when `None` is returned? 
How the flow will behave when `not subscribable_object`?  Will it return `None`?
![image](https://github.com/user-attachments/assets/db2272e3-644f-4098-a15d-95d868532448)

Reply:
Added

Reply:
When we fail to get the subscribable object the payload would be None. In such case `send_webhook_request_async` will not deliver the webhook, the delivery and attempt objects will be saved with "failed" status.